zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/SearchQueryRequestMapper.java

```diff
@@ -138,11 +146,131 @@ public static Either<ProblemDetail, ProcessInstanceSort> toSearchQuerySort(
     return Either.right(null);
   }
 
+  public static Either<ProblemDetail, UserTaskSort> toUserTaskSearchQuerySort(
+      final List<SearchQuerySortRequest> sorting) {
+    if (sorting != null && !sorting.isEmpty()) {
+      final var builder = SortOptionBuilders.userTask();
+
+      for (final SearchQuerySortRequest sort : sorting) {
+        final var field = sort.getField();
+        final var order = sort.getOrder();
+
+        if ("creationTime".equals(field)) {
+          builder.creationDate();
+        } else if ("completionTime".equals(field)) {
+          builder.completionDate();
+        } else {
+          throw new RuntimeException("unkown sortBy " + field);
+        }
+
+        if ("asc".equalsIgnoreCase(order)) {
+          builder.asc();
+        } else if ("desc".equalsIgnoreCase(order)) {
+          builder.desc();
+        } else {
+          throw new RuntimeException("unkown sortOrder " + order);
```

Comment:
🔧 Same as above, it might even make sense to collect input validation errors and return them collectively. That's what we do in other user task methods in this class, for example. That way, users get all validation errors in one go and not bit by bit, request by request.